esbuild doesn't like empty styles property on a component

### Command

build

### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

_No response_

### Description

adding a property to the component decorator with an empty string in the style array causes this error:


```✘ [ERROR] component style name should always be found [css;99;/Users/joshpike/Projects/aquipa/client/src/components/common/clientgroup-invitation-card/clientgroup-invitation-card.component.ts] [plugin angular-css]

    node_modules/@angular-devkit/build-angular/src/builders/browser-esbuild/css-plugin.js:80:42:
      80 │ ...      (0, node_assert_1.default)(data, `component style name sh...
         ╵                                    ^

    at /Users/joshpike/Projects/aquipa/client/node_modules/@angular-devkit/build-angular/src/builders/browser-esbuild/css-plugin.js:80:43
    at requestCallbacks.on-load (/Users/joshpike/Projects/aquipa/client/node_modules/esbuild/lib/main.js:1421:28)
    at handleRequest (/Users/joshpike/Projects/aquipa/client/node_modules/esbuild/lib/main.js:723:19)
    at handleIncomingPacket (/Users/joshpike/Projects/aquipa/client/node_modules/esbuild/lib/main.js:745:7)
    at Socket.readFromStdout (/Users/joshpike/Projects/aquipa/client/node_modules/esbuild/lib/main.js:673:7)
    at Socket.emit (node:events:526:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)

  This error came from the "onLoad" callback registered here:

    node_modules/@angular-devkit/build-angular/src/builders/browser-esbuild/css-plugin.js:78:18:
      78 │             build.onLoad({ filter: /^css;/, namespace: 'angular:st...
         ╵                   ~~~~~~

    at setup (/Users/joshpike/Projects/aquipa/client/node_modules/@angular-devkit/build-angular/src/builders/browser-esbuild/css-plugin.js:78:19)
    at handlePlugins (/Users/joshpike/Projects/aquipa/client/node_modules/esbuild/lib/main.js:1279:21)
    at buildOrContextImpl (/Users/joshpike/Projects/aquipa/client/node_modules/esbuild/lib/main.js:968:5)
    at Object.buildOrContext (/Users/joshpike/Projects/aquipa/client/node_modules/esbuild/lib/main.js:776:5)
    at /Users/joshpike/Projects/aquipa/client/node_modules/esbuild/lib/main.js:2163:15
    at new Promise (<anonymous>)
    at Object.build (/Users/joshpike/Projects/aquipa/client/node_modules/esbuild/lib/main.js:2162:25)
    at build (/Users/joshpike/Projects/aquipa/client/node_modules/esbuild/lib/main.js:2011:51)
    at BundlerContext.bundle (/Users/joshpike/Projects/aquipa/client/node_modules/@angular-devkit/build-angular/src/builders/browser-esbuild/esbuild.js:70:52)
```


### Minimal Reproduction

This below diff shows the offending line of code, and its removal fixed the problem
![Screenshot 2023-03-31 at 11 51 02 am](https://user-images.githubusercontent.com/9585787/229025641-6ad8300c-8618-4f5b-be0f-a047fc8be28d.png)


### Exception or Error

_No response_

### Your Environment

```text
angular 16-beta-4
```


### Anything else relevant?

_No response_

--- a/packages/angular_devkit/build_angular/src/builders/browser-esbuild/css-plugin.ts
+++ b/packages/angular_devkit/build_angular/src/builders/browser-esbuild/css-plugin.ts
@@ -80,7 +80,10 @@ export function createCssPlugin(options: CssPluginOptions): Plugin {
       // Add a load callback to support inline Component styles
       build.onLoad({ filter: /^css;/, namespace: 'angular:styles/component' }, async (args) => {
         const data = options.inlineComponentData?.[args.path];
-        assert(data, `component style name should always be found [${args.path}]`);
+        assert(
+          typeof data === 'string',
+          `component style name should always be found [${args.path}]`,
+        );
 
         const [, , filePath] = args.path.split(';', 3);
 

--- a/packages/angular_devkit/build_angular/src/builders/browser-esbuild/less-plugin.ts
+++ b/packages/angular_devkit/build_angular/src/builders/browser-esbuild/less-plugin.ts
@@ -40,7 +40,10 @@ export function createLessPlugin(options: LessPluginOptions): Plugin {
       // Add a load callback to support inline Component styles
       build.onLoad({ filter: /^less;/, namespace: 'angular:styles/component' }, async (args) => {
         const data = options.inlineComponentData?.[args.path];
-        assert(data, `component style name should always be found [${args.path}]`);
+        assert(
+          typeof data === 'string',
+          `component style name should always be found [${args.path}]`,
+        );
 
         const [, , filePath] = args.path.split(';', 3);
 

--- a/packages/angular_devkit/build_angular/src/builders/browser-esbuild/sass-plugin.ts
+++ b/packages/angular_devkit/build_angular/src/builders/browser-esbuild/sass-plugin.ts
@@ -64,7 +64,10 @@ export function createSassPlugin(options: SassPluginOptions): Plugin {
 
       build.onLoad({ filter: /^s[ac]ss;/, namespace: 'angular:styles/component' }, async (args) => {
         const data = options.inlineComponentData?.[args.path];
-        assert(data, `component style name should always be found [${args.path}]`);
+        assert(
+          typeof data === 'string',
+          `component style name should always be found [${args.path}]`,
+        );
 
         const [language, , filePath] = args.path.split(';', 3);
         const syntax = language === 'sass' ? 'indented' : 'scss';

--- a/packages/angular_devkit/build_angular/src/builders/browser-esbuild/tests/behavior/component-stylesheets_spec.ts
+++ b/packages/angular_devkit/build_angular/src/builders/browser-esbuild/tests/behavior/component-stylesheets_spec.ts
@@ -1,0 +1,27 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { buildEsbuildBrowser } from '../../index';
+import { BASE_OPTIONS, BROWSER_BUILDER_INFO, describeBuilder } from '../setup';
+
+describeBuilder(buildEsbuildBrowser, BROWSER_BUILDER_INFO, (harness) => {
+  describe('Behavior: "Component Stylesheets"', () => {
+    it('should successfuly compile with an empty inline style', async () => {
+      await harness.modifyFile('src/app/app.component.ts', (content) => {
+        return content.replace('styleUrls', 'styles').replace('./app.component.css', '');
+      });
+
+      harness.useTarget('build', {
+        ...BASE_OPTIONS,
+      });
+
+      const { result } = await harness.executeOnce();
+      expect(result?.success).toBeTrue();
+    });
+  });
+});

